python/GafferUI/TabbedContainer.py

```diff
@@ -266,6 +275,29 @@ def __init__( self, parent = None ) :
 
 		QtWidgets.QTabWidget.__init__( self, parent )
 
+		# We have to track tab-moves and propagate them
+		# to our parent TabbedContainer so it can re-order
+		# it's widget list
+		self.__weakTabMoved = Gaffer.WeakMethod( self.__tabMoved )
+		self.tabBar().tabMoved.connect( self.__weakTabMoved )
+		self._tabMovedSignal = None
+
+	def tabMovedSignal( self ) :
+
+		if self._tabMovedSignal is None :
```

Comment:
The goal was to ensure it was stable, even if some other derived class called `setTabBar` in the future. `TabbedContainer` is the only place at the moment that this happens, but wondered if there was the possibility of client code doing this in custom UIs so opted to be on the safe side.

Reply:
> The goal was to ensure it was stable, even if some other derived class called setTabBar in the future. TabbedContainer is the only place at the moment that this happens, but wondered if there was the possibility of client code doing this in custom UIs so opted to be on the safe side.

Any safety here is illusory. If something else calls `setTabBar()`, the Qt tab bar will then be out of sync with the `TabbedContainer.__tabBar` attribute holding the `GafferUI.Widget` part of it. The tab bar is the private domain of the `TabbedContainer`, and by suggesting otherwise we're just misleading the reader and inviting bugs.

Reply:
Ahh yes - of course!

Reply:
Fixed in https://github.com/GafferHQ/gaffer/pull/3167/commits/ba718472a44bb5d1685392a30290d96e265c74a1